apps/web/vibes/soul/sections/reviews/index.tsx

```diff
@@ -1,41 +1,76 @@
+import { Suspense } from 'react'
+
+import { Streamable } from '@/vibes/soul/lib/streamable'
+import { mapStreamable } from '@/vibes/soul/lib/streamable/server'
 import { CursorPagination, CursorPaginationInfo } from '@/vibes/soul/primitives/cursor-pagination'
 import { Rating } from '@/vibes/soul/primitives/rating'
 import { StickySidebarLayout } from '@/vibes/soul/sections/sticky-sidebar-layout'
 
+interface Review {
+  id: string
+  rating: number
+  review: string
+  name: string
+  date: string
+}
+
 interface Props {
-  reviews: {
-    id: string
-    rating: number
-    review: string
-    name: string
-    date: string
-  }[]
-  averageRating: number
-  totalCount?: number
-  paginationInfo?: CursorPaginationInfo
+  reviews: Streamable<Review[]>
+  averageRating: Streamable<number>
+  totalCount?: Streamable<number>
+  paginationInfo?: Streamable<CursorPaginationInfo>
   emptyStateMessage?: string
+  reviewsLabel?: string
 }
 
-export function Reviews({
-  reviews,
-  averageRating,
-  totalCount,
-  paginationInfo,
+async function ReviewsImpl({
+  reviews: streamableReviews,
+  averageRating: streamableAverageRating,
+  totalCount: streamableTotalCount,
+  paginationInfo: streamablePaginationInfo,
   emptyStateMessage,
+  reviewsLabel = 'Reviews',
 }: Readonly<Props>) {
+  const reviews = await mapStreamable(streamableReviews, resolvedReviews => resolvedReviews)
```

Comment:
I tried to wrap this in a `<Suspense />`, but below we use this value to know if we should render the empty state, so unfortunately I need to resolve here. Any ideas?

Reply:
If we're using `await` there's no need to use `mapStreamable` since we can just do this:

```ts
const reviews = await streamableReviews
```

That being said, using `mapStreamable` should remove the need to have a `ReviewsImpl` and another `Review` component that wraps it

Reply:
You could do something like this:

```ts
return (
  <Suspense fallback={<Skeleton />}>
    {mapStreamable(streamableReview, reviews => reviews.length === 0 ? <EmptyState /> : <ReviewsList reviews={reviews} />)}
  </Suspense>
)
```

What you have right now works completely fine, though, and I'm not too worried about the VIBES code being perfect as much as I am about the pattern itself being close to what we need to support

Reply:
I get it now 👍 . Let me push a quick refactor.